File: decompiled_melonDS/sources/me/magnum/melonds/ui/settings/fragments/SaveFilesPreferencesFragment.java
package me.magnum.melonds.ui.settings.fragments;

import android.os.Bundle;
import androidx.preference.Preference;
import me.magnum.melonds.R;
import me.magnum.melonds.ui.settings.preferences.StoragePickerPreference;
import yb.n;
/* compiled from: r8-map-id-68a076d6c8e19c01d0388d4729d627070c4f7cff1f6777b7d5e4497c81f45b39 */
/* loaded from: classes.dex */
public final class SaveFilesPreferencesFragment extends Hilt_SaveFilesPreferencesFragment implements ci.l {
    public final n Y = new n(new e(this, 4));
    public sd.i Z;

    /* renamed from: b0  reason: collision with root package name */
    public sd.d f9604b0;

    @Override // ci.l
    public final String getTitle() {
        String string = getString(R.string.category_save_files);
        string.getClass();
        return string;
    }

    @Override // androidx.preference.a0
    public final void onCreatePreferences(Bundle bundle, String str) {
        setPreferencesFromResource(R.xml.pref_save_files, str);
        Preference findPreference = findPreference("sram_dir");
        findPreference.getClass();
        ((ci.k) this.Y.getValue()).b((StoragePickerPreference) findPreference);
    }
}

File: app/src/main/java/me/magnum/melonds/ui/settings/fragments/SaveFilesPreferencesFragment.kt
package me.magnum.melonds.ui.settings.fragments

import android.net.Uri
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.preference.PreferenceFragmentCompat
import dagger.hilt.android.AndroidEntryPoint
import me.magnum.melonds.R
import me.magnum.melonds.common.DirectoryAccessValidator
import me.magnum.melonds.common.UriPermissionManager
import me.magnum.melonds.impl.SettingsBackupManager
import me.magnum.melonds.extensions.addOnPreferenceChangeListener
import me.magnum.melonds.ui.settings.PreferenceFragmentHelper
import me.magnum.melonds.ui.settings.PreferenceFragmentTitleProvider
import me.magnum.melonds.ui.settings.preferences.StoragePickerPreference
import javax.inject.Inject

@AndroidEntryPoint
class SaveFilesPreferencesFragment : BasePreferenceFragment(), PreferenceFragmentTitleProvider {

    private val helper by lazy { PreferenceFragmentHelper(this, uriPermissionManager, directoryAccessValidator) }
    @Inject lateinit var uriPermissionManager: UriPermissionManager
    @Inject lateinit var directoryAccessValidator: DirectoryAccessValidator
    @Inject lateinit var settingsBackupManager: SettingsBackupManager

    override fun getTitle() = getString(R.string.category_save_files)

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.pref_save_files, rootKey)
        findPreference<StoragePickerPreference>("sram_dir")?.let { pref ->
            helper.bindPreferenceSummaryToValue(pref)
            helper.setupStoragePickerPreference(pref) { uri, persistDirectory ->
                handleSettingsMirror(uri, persistDirectory)
            }
        }

        val customDirPref = findPreference<StoragePickerPreference>("save_state_custom_dir")
        if (customDirPref != null) {
            helper.bindPreferenceSummaryToValue(customDirPref)
            helper.setupStoragePickerPreference(customDirPref) { _, persistDirectory ->
                persistDirectory()
            }
        }

        findPreference<androidx.preference.ListPreference>("save_state_location")?.let { pref ->
            helper.bindPreferenceSummaryToValue(pref)
            customDirPref?.isVisible = (pref.value == "custom_dir")
            pref.addOnPreferenceChangeListener { _, newValue ->
                customDirPref?.isVisible = (newValue == "custom_dir")
                true
            }
        }
    }

    private fun handleSettingsMirror(uri: Uri, persistDirectory: () -> Unit) {
        if (!settingsBackupManager.isMirrorEnabled()) {
            persistDirectory()
            return
        }

        if (!settingsBackupManager.hasMirrorAt(uri)) {
            persistDirectory()
            settingsBackupManager.requestMirrorWrite()
            return
        }

        AlertDialog.Builder(requireContext())
            .setTitle(R.string.settings_mirror_detected_title)
            .setMessage(R.string.settings_mirror_detected_message)
            .setPositiveButton(R.string.settings_mirror_restore) { _, _ ->
                settingsBackupManager.restoreMirrorFrom(uri)
                persistDirectory()
                settingsBackupManager.requestMirrorWrite()
            }
            .setNegativeButton(R.string.settings_mirror_ignore) { _, _ ->
                persistDirectory()
                settingsBackupManager.overwriteMirrorAt(uri)
                settingsBackupManager.requestMirrorWrite()
            }
            .show()
    }
}
